openedx/core/djangolib/django_plugins.py

```diff
@@ -0,0 +1,360 @@
+            settings_config = _get_settings_config(app_config, project_type, settings_type)
+            if settings_config is None:
+                log.info(
+                    u'Plugin Apps [Settings]: Did NOT find %s for %s and %s',
+                    app_config.name,
+                    project_type,
+                    settings_type,
+                )
+                continue
+
+            plugin_settings_path = _get_module_path(app_config, settings_config, PluginSettings)
+            log.info(u'Plugin Apps [Settings]: Found %s for %s and %s', app_config.name, project_type, settings_type)
+            yield import_module(plugin_settings_path)
+
+    @classmethod
+    def _iter_installable_urls(cls, project_type):
+        """
+        Yields the module path and configuration for Plugin URLs registered for
+        the given project_type.
+        """
+        for app_config in cls._get_app_configs(project_type):
+            url_config = _get_url_config(app_config, project_type)
+            if url_config is None:
+                log.info(u'Plugin Apps [URLs]: Did NOT find %s for %s', app_config.name, project_type)
+                continue
+
+            urls_module_path = _get_module_path(app_config, url_config, PluginURLs)
+            url_config[PluginURLs.namespace] = url_config.get(PluginURLs.namespace, app_config.name)
+            log.info(
+                u'Plugin Apps [URLs]: Found %s with namespace %s for %s',
+                app_config.name,
+                url_config[PluginURLs.namespace],
+                project_type,
+            )
+            yield urls_module_path, url_config
+
+    @classmethod
+    def _get_app_configs(cls, project_type):
+        return cls.get_available_plugins(project_type).itervalues()
+
+
+def _get_module_path(app_config, plugin_config, plugin_cls):
+    return u'{package_path}.{module_path}'.format(
+        package_path=app_config.name,
+        module_path=plugin_config.get(plugin_cls.relative_path, plugin_cls.DEFAULT_RELATIVE_PATH),
+    )
+
+
+def _get_settings_config(app_config, project_type, settings_type):
+    plugin_config = getattr(app_config, PLUGIN_APP_CLASS_ATTRIBUTE_NAME, {})
+    settings_config = plugin_config.get(PluginSettings.config, {})
+    project_type_settings = settings_config.get(project_type, {})
+    return project_type_settings.get(settings_type)
+
+
+def _get_url_config(app_config, project_type):
+    plugin_config = getattr(app_config, PLUGIN_APP_CLASS_ATTRIBUTE_NAME, {})
+    url_config = plugin_config.get(PluginURLs.config, {})
+    return url_config.get(project_type)
+
+
+def _iter_uppercase_attributes(obj):
+    def _is_uppercase(val):
+        return val == val.upper() and not val.startswith('_')
+
+    for name in filter(_is_uppercase, dir(obj)):
+        yield name, getattr(obj, name)
+
+
+def _get_url_regex(url_config):
+    regex = url_config.get(PluginURLs.regex)
+    return r'^{}'.format(regex) if regex else r''
```

Comment:
Does the insertion into a raw string actually accomplish anything?  Backslashes will have already been processed as appropriate for the type of string literal that was used in the configuration.

Reply:
good point.  updated.